code/modules/experisci/experiment/handlers/experiment_handler.dm

```diff
@@ -0,0 +1,393 @@
+	if (!proximity_flag || (selected_experiment == null && !(config_flags & EXPERIMENT_CONFIG_ALWAYS_ACTIVE)))
+		return
+	playsound(user, 'sound/machines/buzz-sigh.ogg', 25)
+	to_chat(user, "<span>\the [target.name] is not related to your currently selected experiment.</span>")
+
+/**
+ * Checks that an experiment can be run using the provided target, used for preventing the cancellation of the attack chain inappropriately
+ */
+/datum/component/experiment_handler/proc/should_run_handheld_experiment(datum/source, atom/target, mob/user, params)
+	// Check that there is actually an experiment selected
+	if (selected_experiment == null && !(config_flags & EXPERIMENT_CONFIG_ALWAYS_ACTIVE))
+		return
+
+	// Determine if this experiment is actionable with this target
+	var/list/arguments = list(src)
+	arguments = args.len > 1 ? arguments + args.Copy(2) : arguments
+	if (config_flags & EXPERIMENT_CONFIG_ALWAYS_ACTIVE)
+		for (var/datum/experiment/e in linked_web.available_experiments)
+			if (e.actionable(arglist(arguments)))
+				return TRUE
+	else
+		. = selected_experiment.actionable(arglist(arguments))
+
+/**
+ * This proc exists because Jared Fogle really likes async
+ */
+/datum/component/experiment_handler/proc/try_run_handheld_experiment_async(datum/source, atom/target, mob/user, params)
+	if (selected_experiment == null && !(config_flags & EXPERIMENT_CONFIG_ALWAYS_ACTIVE))
+		to_chat(user, "<span>You do not have an experiment selected!.</span>")
+		return
+	if(!do_after(user, 10, target = target))
+		return
+	if(action_experiment(source, target))
+		playsound(user, 'sound/machines/ping.ogg', 25)
+		to_chat(user, "<span>You scan \the [target.name].</span>")
+	else
+		playsound(user, 'sound/machines/buzz-sigh.ogg', 25)
+		to_chat(user, "<span>\the [target.name] is not related to your currently selected experiment.</span>")
+
+
+/**
+ * Hooks on destructive scans to try and run an experiment (When using a handheld handler)
+ */
+/datum/component/experiment_handler/proc/try_run_destructive_experiment(datum/source, list/scanned_atoms)
+	SIGNAL_HANDLER
+	var/atom/movable/our_scanner = parent
+	if (selected_experiment == null)
+		playsound(our_scanner, 'sound/machines/buzz-sigh.ogg', 25)
+		to_chat(our_scanner, "<span>No experiment selected!.</span>")
+		return
+	var/successful_scan
+	for(var/scan_target in scanned_atoms)
+		if(action_experiment(source, scan_target))
+			successful_scan = TRUE
+	if(successful_scan)
+		playsound(our_scanner, 'sound/machines/ping.ogg', 25)
+		to_chat(our_scanner, "<span>The scan succeeds.</span>")
+	else
+		playsound(src, 'sound/machines/buzz-sigh.ogg', 25)
+		our_scanner.say("The scan did not result in anything.")
+/**
+ * Hooks on successful explosions on the doppler array this is attached to
+ */
+/datum/component/experiment_handler/proc/try_run_doppler_experiment(datum/source, turf/epicenter, devastation_range,
+	heavy_impact_range, light_impact_range, took, orig_dev_range, orig_heavy_range, orig_light_range)
```

Comment:
Just like with the if statements, it should be obvious where one statement ends and the other begins.

```suggestion
	heavy_impact_range, light_impact_range, took, orig_dev_range, orig_heavy_range, orig_light_range
)
```